packages/openapi3/src/openapi.ts

```diff
@@ -374,8 +374,19 @@ function createOAPIEmitter(program: Program, options: ResolvedOpenAPI3EmitterOpt
     });
   }
 
+  function checkOptionCanonicaliedVersion(program: Program): boolean {
+    if (program.getOption("canonicalized-version") === "true") {
```

Comment:
Those misc options are some legacy behavior and shouldn't be used, please look at this doc on how to specify emitter options https://microsoft.github.io/typespec/extending-typespec/emitters#custom-configuration-options